polaris-react/src/components/DropZone/DropZone.module.scss

```
@@ -59,6 +59,10 @@ $dropzone-stacking-order: (
   &:hover {
     outline: var(--p-border-width-025) solid transparent;
   }
+
+  &.noOutline {
+    background-color: transparent;
+  }
```

You could flip this and then remove the background color from line 35.

```suggestion
  &:not(.noOutline) {
    background-color: var(--p-color-input-bg-surface);
  }
```